tslearn/clustering.py

```diff
@@ -306,17 +302,31 @@ def fit(self, X, y=None, sample_weight=None):
             Weights to be given to time series in the learning process. By default, all time series weights are equal.
         """
 
+        X = check_array(X, allow_nd=True)
```

Comment:
Seems like a very good idea to use `sklearn`'s `check_array`, but I wonder if we could have a dedicated `tslearn` one that could, for example, reshape the data if it is 2d, as done for now for some estimators, see as an example: https://github.com/rtavenar/tslearn/blob/master/tslearn/barycenters.py#L102

And then we could use this dedicated `check_array` in all our estimators.